src/handler/put.ts

```diff
@@ -33,33 +33,54 @@ export const handler = async (event: APIGatewayProxyEvent): Promise<APIGatewayPr
   if (!event.queryStringParameters.manufacturerCampaignReference && !event.queryStringParameters.dvsaCampaignReference) {
     return HttpResponse(StatusCodes.BAD_REQUEST, ExternalApiErrorMessages.DvsaAndManufacturerCampaignReferenceMissing);
   }
-  let recallUpdate: RecallsUpdateRequest;
-  try {
-    recallUpdate = JSON.parse(event.body) as RecallsUpdateRequest;
-    if (!allRequiredFieldsUpdateRecall(recallUpdate)) {
-      return HttpResponse(StatusCodes.BAD_REQUEST, ExternalApiErrorMessages.InvalidRequestBody);
-    }
-  } catch (err) {
-    return HttpResponse(StatusCodes.BAD_REQUEST, ExternalApiErrorMessages.InvalidRequestBody);
-  }
+
   const { vin } = event.pathParameters;
   const { manufacturerCampaignReference, dvsaCampaignReference } = event.queryStringParameters;
+  const recallUpdate = JSON.parse(event.body) as RecallsUpdateRequest;
+  if (recallUpdate.repairStatus == RepairStatus.FIXED) {
+    if (!validDateFormat(recallUpdate.rectificationDate)) {
+      return HttpResponse(StatusCodes.BAD_REQUEST, ExternalApiErrorMessages.InvalidDateFormat);
+    }
+  }
+
   let vehicleFound;
+
   if (dvsaCampaignReference) {
     vehicleFound = Vehicles.find((vehicle) => vehicle.vin === vin && vehicle.dvsaCampaignReference === dvsaCampaignReference);
     logger.info('dvsaCampaignreference', { vehicleFound });
   } else {
     vehicleFound = Vehicles.find((vehicle) => vehicle.vin === vin && vehicle.manufacturerCampaignReference === manufacturerCampaignReference);
     logger.info('manufacturerCampaignreference', { vehicleFound });
   }
-  if (vehicleFound) {
-    if (alreadyRepaired(vehicleFound)) {
-      return HttpResponse(StatusCodes.BAD_REQUEST, ExternalApiErrorMessages.VehicleRecallAlreadyFixed);
+
+  if (!vehicleFound) {
+    return HttpResponse(StatusCodes.NOT_FOUND, getReasonPhrase(StatusCodes.NOT_FOUND));
+  }
+
+  if (vehicleFound.repairStatus === RepairStatus.NOT_FIXED) {
+    if (!allRequiredFieldsUpdateNonfixedRecall(recallUpdate)) {
+      return HttpResponse(StatusCodes.BAD_REQUEST, ExternalApiErrorMessages.InvalidRequestBody);
     }
-    if (!validDateFormat(recallUpdate.rectificationDate)) {
-      return HttpResponse(StatusCodes.BAD_REQUEST, ExternalApiErrorMessages.InvalidDateFormat);
+    if (recallUpdate.repairStatus === RepairStatus.FIXED) {
+      if (vehicleFound) {
```

Comment:
Minor: Logic here is a bit odd to me. First we check `vehicleFound.repairStatus` and later we check `vehicleFound`. In my opinion this line is not needed.

Reply:
changed to remove extra check